File: java/src/test/java/com/bc/ur/TestUtils.java
package com.bc.ur;

public class TestUtils {
    public static Byte[] toTypedArray(byte[] bytes) {
        Byte[] result = new Byte[bytes.length];
        for (int i = 0; i < bytes.length; i++)
            result[i] = bytes[i];
        return result;
    }

    public static String bytes2Hex(byte[] bytes) {
        StringBuilder builder = new StringBuilder();
        for (byte b : bytes) {
            builder.append(String.format("%x", b));
        }
        return builder.toString();
    }
}

File: android/app/src/androidTest/java/com/bc/ur/TestUtils.kt
package com.bc.ur

fun bytes2Hex(bytes: ByteArray): String? {
    val builder = StringBuilder()
    for (b in bytes) {
        builder.append(String.format("%x", b))
    }
    return builder.toString()
}

inline fun <T : AutoCloseable?, R> T.use(block: (T) -> R): R {
    var exception: Exception? = null
    try {
        return block(this)
    } catch (e: Exception) {
        exception = e
        throw e
    } finally {
        when {
            this == null -> {
            }
            exception == null -> close()
            else ->
                try {
                    close()
                } catch (closeException: Exception) {
                    // cause.addSuppressed(closeException) // ignored here
                }
        }
    }
}

inline fun <reified T : Throwable> assertThrows(
    msg: String?,
    callable: () -> Unit
): T {
    try {
        callable()
        throw RuntimeException(msg)
    } catch (e: Throwable) {
        if (e is T) return e
        throw e
    }
}